src/lib/client.ts

```diff
@@ -203,6 +208,6 @@ export class Client {
   }
 }
 
-export function startFarming(path: string): Promise<string> {
-  return (tauri.invoke('farming', { path })).then(value => value as string)
+export async function startFarming(path: string): Promise<ClientIdentity> {
```

Comment:
This will work, but function doesn't need to be `async` in this case since there is no `await` in it.

Reply:
alright, will include in the next pr :)